test/spec/request-test.js

```diff
@@ -1425,4 +1427,263 @@ describe('Request-Promise', function () {
+            }
+        });
+
+        assertResponse = function(response, statusCode, content, lastModified, age) {
+            expect(response.statuCode).to.be.equal(statusCode)
+            expect(JSON.parse(response.body).name).to.be.equal(content)
+            expect(response.headers['last-modified']).to.be.equal(now)
+            if (!age) {
+                expect(response.headers['age']).to.be.equal(age)
+            } else {
+                expect(response.headers['age'] > 0).to.be.true
+            }
+        }
+
+        it('should cache resource : cache-control, cache option true', function (done) {
+            var options = {
+                uri: 'http://localhost:5000/resource1',
+                method: 'GET',
+                cache: true
+            }
+
+            rp(options).then(function(response){
+                assertResponse(response, 200, 'resource1', now, 0)
+                rp(options).then(function(response){
+                    assertResponse(response, 200, 'resource1', now, true)
+                    done()
+                })
+            })
+
+        });
+
+        it('should not cache resource : no cache-control, cache option true', function (done) {
+            var options = {
+                uri: 'http://localhost:5000/resource2',
+                method: 'GET',
+                cache: true
+            }
+
+            rp(options).then(function(response){
+                assertResponse(response, 200, 'resource2', now, 0)
+                rp(options).then(function(response){
+                    assertResponse(response, 200, 'resource2', now, 0)
+                    done()
+                })
+            })
+
+        });
+
+        it('should not cache resource : cache-control no-cache', function (done) {
+            var options = {
+                uri: 'http://localhost:5000/resource3',
+                method: 'GET',
+                cache: true
+            }
+
+            rp(options).then(function(response){
+                assertResponse(response, 200, 'resource3', now, 0)
+                rp(options).then(function(response){
+                    assertResponse(response, 200, 'resource3', now, 0)
+                    done()
+                })
+            })
+
+        });
+
+        it('should not cache resource : cache-control, cache option false', function (done) {
+            var options = {
+                uri: 'http://localhost:5000/resource1',
+                method: 'GET',
+                cache: false
+            }
+
+            rp(options).then(function(response){
+                assertResponse(response, 200, 'resource1', now, 0)
+                rp(options).then(function(response){
+                    assertResponse(response, 200, 'resource1', now, 0)
+                    done()
+                })
+            })
+
+        });
+
+        it('should cache resource and deliver a stale (no invalidation)', function (done) {
```

Comment:
Test case

Assumptions: 
- the resource returns a `cache-control` header with a positive max-age
- the cache is previously empty
- the cache option is true
- the origin resource gets updated
- we request the cache before the max-age is exceeded

results:
- the resource get cached after first request
- the resource that we get after second request is the cached one because the max-age is not exceeded (even if origin resource has been updated)